src/compiler/checker.ts

```diff
@@ -2797,7 +2797,7 @@ namespace ts {
             }
         }
 
-        function isEntityNameVisible(entityName: EntityNameOrEntityNameExpression, enclosingDeclaration: Node): SymbolVisibilityResult {
+        function resolveNameFromEntityName(entityName: EntityNameOrEntityNameExpression, enclosingDeclaration: Node) {

```

Comment:
Don't we already have a `resolveEntityName` function?

Reply:
This skips a lot of the work `resolveEntityName` does to verify namespaceyness and also generates an appropriate meaning from the `entityName`'s parent. So they're a bit different, unfortunately.

Reply:
We either should have only one function with options to control behavior (preferred), or should clearly document the difference between the two.